src/utilities/shared-data-utils.ts

```diff
@@ -12,8 +12,8 @@ export const isLinkedToTile = (model: ITileModel, tileId: string) => {
   const sharedModelManager = getSharedModelManager(model);
   if (sharedModelManager?.isReady) {
     const modelDataSet = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, model.id);
-    const sourceTileDataSet = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, tileId);
-    if (sourceTileDataSet?.id === modelDataSet?.id) {
+    const sharedModelTileIds = sharedModelManager?.getSharedModelTileIds(modelDataSet);
+    if (sharedModelTileIds?.includes(tileId)) {
       return true;
     }
```

Comment:
I found that when a Geometry tile was linked to multiple tiles, the link dialogs would only show that the Geometry tile was linked to the first tile it had been linked to. This fixes that so all linked tiles are listed as linked in the dialogs.